src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/impl/OrcidOAuth2AP.java

```diff
@@ -55,7 +55,12 @@
     public static final String PROVIDER_ID_SANDBOX = "orcid-sandbox";
     
     public OrcidOAuth2AP(String clientId, String clientSecret, String userEndpoint) {
-        scope = Arrays.asList("/read-limited");
+
+        String s = null;
+        if(userEndpoint != null){
+            s = userEndpoint.startsWith("https://pub") ? "/authenticate" : "/read-limited";
+        }
+        scope = Arrays.asList(s);
```

Comment:
One could argue that this can be done more compact with only one `if` and without an extra variable allocation.

E. g. like
```java
if(userEndpoint != null && userEndpoint.startsWith("https://pub")) {
    this.scope = Arrays.asList("/authenticate");
} else {
    this.scope = Arrays.asList("/read-limited");
}
```

Or you leave it to the code optimizer of the JVM... :wink:

Reply:
I'm fine with letting the JVM optimize it. 😄